### PR TITLE
Refactoring: Modular monolith architecture

### PR DESCRIPTION
**Pull request by senghe**

### CODE REVIEW

File: src/Component/Core/Admin/Resources/routing/product_listing.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+open_marketplace_admin_product_listing_reject:
+    path: /product-listings/{id}/reject
+    defaults:
+        _controller: bitbag.open_marketplace.component.core.admin.controller.product_listing.reject
+        _sylius:
+            section: admin
+            permission: true
+
+open_marketplace_admin_product_listing_restore:
+    path: /product-listings-restore/{id}
```
Comment (maxbeckers) [inline src/Component/Core/Admin/Resources/routing/product_listing.yaml:27]: Make it an action resource like the other routes
```suggestion
    path: /product-listings/{id}/restore
```

File: src/Component/Core/Common/Controller/Resource/VendorController.php [hunk 2]
```diff
@@ -45,30 +45,26 @@
...
+        protected ResourceFormFactoryInterface $resourceFormFactory,
+        protected RedirectHandlerInterface $redirectHandler,
+        protected FlashHelperInterface $flashHelper,
+        protected AuthorizationCheckerInterface $authorizationChecker,
+        protected EventDispatcherInterface $eventDispatcher,
+        protected ?StateMachineInterface $stateMachine,
+        protected ResourceUpdateHandlerInterface $resourceUpdateHandler,
+        protected ResourceDeleteHandlerInterface $resourceDeleteHandler,
+        protected VendorContextInterface $vendorProvider,
+        protected ProfileUpdaterInterface $vendorProfileUpdater
```
Comment (maxbeckers) [inline src/Component/Core/Common/Controller/Resource/VendorController.php:67]: Why are they all protected and not private?

File: src/Component/Core/Common/Fixture/Factory/ConversationExampleFactory.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        $this->optionsResolver = new OptionsResolver();
+
+        $this->configureOptions($this->optionsResolver);
+    }
+
+    protected function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver
+            ->setDefault('number_of_messages', fn (Options $options): int => random_int(4, 10))
+
```
Comment (maxbeckers) [inline src/Component/Core/Common/Fixture/Factory/ConversationExampleFactory.php:54]: do we need this empty lines?

File: src/Component/Core/Common/Resources/services/fixtures/factories.xml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+Do you need more information about us and what we do? Visit our https://bitbag.io website!
+We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+-->
+
+<container xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://symfony.com/schema/dic/services"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    <services>
+        <defaults public="false" autowire="false" autoconfigure="false" />
+
+
```
Comment (maxbeckers) [inline src/Component/Core/Common/Resources/services/fixtures/factories.xml:14]: only one empty line?

File: src/Component/Core/Vendor/Form/Type/ProductListing/DraftTaxonAutocompleteChoiceType.php [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                    new ProductDraftTaxonToTaxonTransformer(
+                        $this->productDraftTaxonFactory,
+                        $this->productDraftTaxonRepository,
+                        $options['productDraft'],
+                    ),
+                ),
+            );
+        }
+
+        if (!$options['multiple']) {
```
Comment (maxbeckers) [inline src/Component/Core/Vendor/Form/Type/ProductListing/DraftTaxonAutocompleteChoiceType.php:50]: else?

File: src/Component/Core/Vendor/Resources/routing/attributes.yaml [hunk 6]
```diff
@@ -0,0 +1,77 @@
+open_marketplace_vendor_attributes_index:
+    path: /product-attributes/index
```
Comment (maxbeckers) [inline src/Component/Core/Vendor/Resources/routing/attributes.yaml:2]: Maybe this?
```suggestion
    path: /product-attributes
```

File: src/Component/Core/Vendor/Resources/routing/attributes.yaml [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+            vars:
+                subheader: sylius.ui.manage_attributes_of_your_products
+                templates:
+                    form: "@SyliusAdmin/ProductAttribute/_form.html.twig"
+                route:
+                    parameters:
+                        type: $type
+
+open_marketplace_product_draft_attribute_update:
+    path: /product-attributes/{id}/update
```
Comment (maxbeckers) [inline src/Component/Core/Vendor/Resources/routing/attributes.yaml:32]: why is this not a `PATCH` on `/product-attributes/{id}`?

File: src/Component/Core/Vendor/Resources/routing/attributes.yaml [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+                    form: "@SyliusAdmin/ProductAttribute/_form.html.twig"
+                route:
+                    parameters:
+                        id: $id
+
+open_marketplace_vendor_attributes_update:
+    alias: open_marketplace_product_draft_attribute_update
+
+open_marketplace_product_draft_attribute_delete:
+    path: /product-attributes/{id}/delete
```
Comment (maxbeckers) [inline src/Component/Core/Vendor/Resources/routing/attributes.yaml:49]: And this a `DELETE` on `/product-attributes/{id}`

File: src/Component/Core/Vendor/Resources/routing/messaging.yaml [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    methods: [GET, POST]
+    defaults:
+        _controller: bitbag.open_marketplace.component.core.common.controller.messaging.create_message
+        _sylius:
+            redirect: open_marketplace_vendor_messaging_conversation_show
+            mail_redirect: open_marketplace_vendor_messaging_conversation_show
+
+open_marketplace_vendor_messaging_conversation_create:
+    path: /conversation/create
+    methods: [GET,POST]
```
Comment (maxbeckers) [inline src/Component/Core/Vendor/Resources/routing/messaging.yaml:28]: GET on a create action sounds stupid
```suggestion
    path: /conversations
    methods: [POST]
```

File: src/Component/Core/Common/Fixture/Factory/ConversationExampleFactory.php (hunk 3 above)
Comment (senghe) [reply src/Component/Core/Common/Fixture/Factory/ConversationExampleFactory.php:54]: Don't think so

File: src/Component/Core/Vendor/Resources/routing/attributes.yaml (hunk 8 above)
Comment (senghe) [reply src/Component/Core/Vendor/Resources/routing/attributes.yaml:49]: It is used as a form-target in GUI, so it shouldn't be DELETE in my opinion

File: src/Component/Core/Vendor/Resources/routing/messaging.yaml (hunk 9 above)
Comment (senghe) [reply src/Component/Core/Vendor/Resources/routing/messaging.yaml:28]: It's also used in GUI as GET endpoint

**Pull request merged**